src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -354,121 +361,138 @@ int ACA_Dataplane_OVS::update_neighbor_state_workitem(NeighborState current_Neig
 
   switch (current_NeighborState.operation_type()) {
   case OperationType::CREATE:
+    [[fallthrough]];
+  case OperationType::UPDATE:
+    [[fallthrough]];
+  case OperationType::INFO:
 
-    // TODO: add support for more than one fixed_ips in the future
-    assert(current_NeighborConfiguration.fixed_ips_size() == 1);
+    if (current_NeighborConfiguration.fixed_ips_size() <= 0) {
+      ACA_LOG_ERROR("NeighborConfiguration.fixed_ips_size: %d.\n",
+                    current_NeighborConfiguration.fixed_ips_size());
+      throw std::invalid_argument("NeighborConfiguration.fixed_ips_size is less than zero");
+    }
 
-    if (current_NeighborConfiguration.fixed_ips(0).neighbor_type() == NeighborType::L2 ||
-        current_NeighborConfiguration.fixed_ips(0).neighbor_type() == NeighborType::L3) {
-      try {
-        assert(current_NeighborConfiguration.fixed_ips_size() == 1);
-        virtual_ip_address = current_NeighborConfiguration.fixed_ips(0).ip_address();
+    for (int ip_index = 0;
+         ip_index < current_NeighborConfiguration.fixed_ips_size(); ip_index++) {
+      if (current_NeighborConfiguration.fixed_ips(ip_index).neighbor_type() == NeighborType::L2 ||
```

Comment:
As we use current_NeighborConfiguration.fixed_ips(ip_index) for more than once, recommend to use a local variable to store it.

Reply:
fixed, that's a good change.